subprojects/core/src/main/java/org/gradle/internal/service/scopes/ExecutionGlobalServices.java

```diff
@@ -156,20 +148,17 @@ TypeAnnotationMetadataStore createAnnotationMetadataStore(CrossBuildInMemoryCach
                 "kotlin"
             ),
             ImmutableSet.of(
-                ClosureBackedAction.class,
+                // Getter called when there is a nested bean with action in a Task
                 ConfigureUtil.WrappedConfigureAction.class,
-                Describable.class,
+                // Inputs can extend these classes,
+                // so we want to ignore get methods declared in these classes
                 DefaultDomainObjectCollection.class,
-                DefaultDomainObjectSet.class,
                 DefaultNamedDomainObjectCollection.class,
-                DefaultNamedDomainObjectList.class,
                 DefaultNamedDomainObjectSet.class,
-                DefaultPolymorphicDomainObjectContainer.class,
-                DynamicObjectAware.class,
+                // Task extends this interface
                 ExtensionAware.class,
```

Comment:
How does this work actually? If a task has an extension, shouldn't that be an input to it?

Reply:
`ExtensionContainer getExtensions()` is `@Internal` on `AbstractTask`. Should check what happens when you change extension value, but I don't think this was ever meant as input.

Reply:
There are two questions:

1) Since `getExtensions()` is `@Internal` on `AbstractTask`, can we remove the type from here? 
2) Should we revisit this idea that extensions are not considered inputs? I'll raise this on Slack.

Reply:
With 1. it fails atm. We need to annotate it on Task itself

Reply:
I figured why it fails: some Kotlin task interfaces implement `Task` directly, so we scan Task